stellargraph/core/element_data.py

```diff
@@ -303,6 +305,37 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class FlatAdjacencyList:
+    def __init__(self, flat_array, neighbor_counts):
+        self.splits = np.cumsum(np.append(0, neighbor_counts))
+        self.max_node_iloc = len(neighbor_counts) - 1
+        self.flat = flat_array
+        self.empty = np.array([], dtype=np.uint8)
+
+    def __getitem__(self, idx):
+        start, stop = self.splits[idx], self.splits[idx + 1]
```

Comment:
I strongly suspect this will be faster as just:

```suggestion
        start = self.splits[idx]
        stops = self.splits[idx + 1]
```

although I suspect it may not be actually be noticeable.

Reply:
weirdly I think this is very very slightly slower:

```python
t = (10, 20)
def f():
    a, b = t

%time _ = [f() for _ in range(1000)]  
```

Takes 170us, but:

```python
t = (10, 20)
def f():
    a = t[0]
    b = t[1]

%time _ = [f() for _ in range(1000)]  
```

takes 216us

I know, I know, I should be using `timeit` but I got impatient

Reply:
oh nice! i didn't know about `dis`, thanks for the explanation :)